### ISSUE TITLE
kuksa adapter should convert value from platform type (e.g. number) to string to fit to the VISS V2 standard

### ISSUE THREAD

**Issue by wenwenchenbosch** [GENIVI/iot-event-analytics#109]
Currently, subscribed value from `kuksa.val` is not published to talents, if define the corresponding platform type is defined to number, because the type of responded value from  `kuksa.val` is always string. This is also defined in the VISS v2 standard: 

VISS V2 defined, that value to be set and read should be string. see https://raw.githack.com/w3c/automotive/gh-pages/spec/VISSv2_Transport.html#term-def

Attribute | Type | Description
-- | -- | --
value | string | The data value associated with the path.

Changing the type from number to string here works, but according to the `datatype` of vss model, it should be number (int32).
https://github.com/GENIVI/iot-event-analytics/blob/809993fa1a62b42d5afb9c95ec207f6094844d7c/src/sdk/javascript/examples/integrations/kuksa.val/config/platform/types.json#L19
Talked with @JochenKienzle, a type converter inside kuksa adapter will be implemented. The `datatype` of metadata should be then considered for convention.  

Additional, `kuksa.val` accept now also non-string type for `setRequest`, which is not standard confirm. Therefore, the behavior will be changed in the future. An issue is already create for it: https://github.com/eclipse/kuksa.val/issues/209

**Comment by llakie** [comment]
I do not get this "feature". If it's defined in the vss.json, that the value is a number, why do I get a `string` then? The same goes for all other value types but string. In my opinion this just generates overhead for de-/serializing, which has to be handled by the consumer. Therefore it's necessary, that the consumer actually knows the datatype from some kind of metadata source (additional request) and has to parse it. Kuksa.Val also needs to parse/validate the given string to make sure it complies to the underlying vss.json.

**Comment by wenwenchenbosch** [comment]
yes, it is a valid point. @SebastianSchildt can you give some reason, why `string` is defined in the standard? Just to say, it is so defined is unconvincing.

**Comment by SebastianSchildt** [comment]
probably @UlfBj  can answer it

I assume because not all VSS datatypes have a clear JSON counterpart (i.e. int8,16,32, float vs. double...), so just using string is cleaner/type checking is needed anyway. But just guessing here.

**Comment by llakie** [comment]
I agree, that JSON just supports numbers [+string, boolean, array, object] and thus additional checking is mandatory. But clamping or testing out upper lower bounds etc. for numbers can be done using these plain numbers. Transforming strings to numbers and vice versa is actually an additional step. I don't find it cleaner to use strings. I always prefer e.g. comparing boolean to boolean, numbers with numbers... since I know many cases in where people are discussing whether "1", "true", "True", "TRUE", "yes" should be translated to boolean true. That's a pitfall which comes by using stringified values. The same applies to numbers (using "," or "." as a thousand or decimal separator because of English or German is a nightmare to parse)
And if they want to use JSON as their representation of choice, they have to deal with the "limitations" of this format and not making everything on top super complicated to compensate for that. In the end, the whole thing gets complicated without no reason. In addition, strings tend to be misused for "weirdo-datatypes": First byte means this, second bytes this and then, separated by an ampersand there's the value. And don't even try to do another encoding than UTF-8, otherwise you break it.
I know, it's a bit provoking, but I hope you get my point.
